packages/cli/src/lib/build.ts

```diff
@@ -1,67 +1,99 @@
 import {fileURLToPath} from 'node:url';
+import {findPathUp} from '@shopify/cli-kit/node/fs';
+import {AbortError} from '@shopify/cli-kit/node/error';
+import {joinPath} from '@shopify/cli-kit/node/path';
 import {execAsync} from './process.js';
 
-export const GENERATOR_TEMPLATES_DIR = 'generator-templates';
-export const GENERATOR_STARTER_DIR = 'starter';
-export const GENERATOR_APP_DIR = 'app';
-export const GENERATOR_ROUTE_DIR = 'routes';
-export const GENERATOR_SETUP_ASSETS_DIR = 'assets';
-export const GENERATOR_SETUP_ASSETS_SUB_DIRS = [
-  'tailwind',
-  'css-modules',
-  'vanilla-extract',
-  'postcss',
-  'vite',
-] as const;
-
-export type AssetDir = (typeof GENERATOR_SETUP_ASSETS_SUB_DIRS)[number];
-
-export function getAssetDir(feature: AssetDir) {
-  if (process.env.NODE_ENV === 'test') {
-    return fileURLToPath(
-      new URL(`../setup-assets/${feature}`, import.meta.url),
+// Avoid using fileURLToPath here to prevent backslashes nightmare on Windows
+const monorepoPackagesPath = new URL('../../..', import.meta.url).pathname;
+export const isHydrogenMonorepo = monorepoPackagesPath.endsWith(
+  '/hydrogen/packages/',
+);
+export const hydrogenPackagesPath = isHydrogenMonorepo
+  ? monorepoPackagesPath
+  : undefined;
+
+// The global CLI will merge our assets with other
+// plugins so we must namespace the directory:
+export const ASSETS_DIR_PREFIX = 'assets/hydrogen';
+
+export const ASSETS_STARTER_DIR = 'starter';
+export const ASSETS_STARTER_DIR_ROUTES = 'routes';
+
+export type AssetsDir =
+  | 'tailwind'
+  | 'css-modules'
+  | 'vanilla-extract'
+  | 'postcss'
+  | 'vite'
+  | 'i18n'
+  | 'routes'
+  | 'bundle'
+  // These are created at build time:
+  | 'virtual-routes'
+  | typeof ASSETS_STARTER_DIR;
+
+let pkgJsonPath: string | undefined;
+export async function getAssetsDir(feature?: AssetsDir, ...subpaths: string[]) {
+  pkgJsonPath ??= await findPathUp('package.json', {
+    cwd: fileURLToPath(import.meta.url),
+    type: 'file',
+  });
+
+  if (!pkgJsonPath) {
+    throw new AbortError(
+      'Could not find assets directory',
+      'Please report this error.',
     );
   }
 
-  return fileURLToPath(
-    new URL(
-      `./${GENERATOR_TEMPLATES_DIR}/${GENERATOR_SETUP_ASSETS_DIR}/${feature}`,
-      import.meta.url,
-    ),
+  return joinPath(
+    pkgJsonPath.replace(/\/package\.json$/, ''),
```

Comment:
instead of replace, you can use `dirname(pkgJsonPath)` to get the directory where the assets are